src/playlist.js

```diff
@@ -72,10 +77,13 @@ export const getHoldBack = (master, media) => {
   } else if (media.targetDuration) {
     // TODO: this should probably be targetDuration * 3
     // but we use this for backwards compatability.
-    const lastPartSegment = partSegments[partSegments.length - 1];
+    const lastPartSegment = partsAndSegments[partsAndSegments.length - 1];
     const lastPartDuration = lastPartSegment && lastPartSegment.duration || media.targetDuration;
 
     return lastPartDuration + media.targetDuration * 2;
+  // we shouldn't ever end up using lastThreeDurations as targetDuration
+  // is usually required, but if we somehow get here, with a missing
+  // targetDuration we should handle it
   } else if (lastThreeDurations) {
     return lastThreeDurations;
   }
```

Comment:
Since we shouldn't end up here, it's probably worth pulling all of that logic down into an `else` here and return either 0 or whatever we calculate as the last three. Might also be worth warning (if we don't warn elsewhere for missing target duration), or logging anyway to help us determinelogic in the event of a missing target duration.

Reply:
We use lastThreeDurations in the case where we have parts, but no partTargetDuration so that we get a more accurate hold back for ll-hls. We warn about target duration via the manifest parser warnings that now go to the debug log.